src/core/parser/project.ts

```diff
@@ -0,0 +1,22 @@
+import { parseComponentJSON } from './component'
+
+import { ProjectUIDL, ComponentUIDL } from '../../typings/uidl-definitions'
+
+export const parseProjectJSON = (input: Record<string, unknown>): ProjectUIDL => {
+  const root = input.root as ComponentUIDL
+  const components = (input.components || {}) as Record<string, ComponentUIDL>
+
+  const result = {
```

Comment:
the resolver uses a `cloneObject` function at some point to avoid reference leaking and to ensure the input UIDL is not modified by accident. should we move that here now that this is the first step where we potentially change the input?

Reply:
will do

Reply:
It's unclear if we should clone in components as well as in the project. If I add a clone in both place, we will clone twice, once in project and once in component 👎

Reply:
maybe we can put it behind the skipValidation flag? or use a skipParsing flag?

parsing is in a way part of the pre-processing of the UIDL so we can use a single flag for both. no idea which is the best option for now